main.py

```diff
@@ -1,13 +1,144 @@
+                if val in ("1", "2", "3", "4", "5", "6", "7", "8", "9"):
+                    state.user_select_space = int(val)
+                    term_info[2] = (
+                        f"Current: SubGrid {state.user_select_subgrid} "
+                        f"| Space {state.user_select_space}"
+                    )
+
+                print(term.move_up(7))
+                user_section = update_user_section(term_info)
+                print("".join(user_section))
+
+            if val.is_sequence and val.name == "KEY_ENTER":
+                if state.subgrid_select and state.user_select_subgrid != 0:
+                    state.subgrid_select = False
+                    term_info[1] = "Select Space by entering 1-9"
+                elif state.space_select and state.user_select_space != 0:
+                    state.space_select = False
+                    term_info[1] = "Confirm Selection?"
+                elif (
+                    state.user_confirm
+                    and state.user_select_subgrid != 0
+                    and state.user_select_space != 0
+                ):
+                    state.user_confirm = False
+                    # TODO execute game logic here
+                    # update_arrays()
+                    #
+
+                    # TODO update the game board here
+
+                    # show confirmation
+                    term_info[0] = " "
+                    term_info[1] = f"Player{player_active} selected:"
+                    term_info[
+                        2
+                    ] = f"SubGrid {state.user_select_subgrid} | Space {state.user_select_space}"
+
+                    # confirm end of turn
+                    state.end_of_turn = True
+
+                print(term.move_up(7))
+                user_section = update_user_section(term_info)
+                print("".join(user_section))
+
+            if state.end_of_turn:
+                # delay for user to read confirmation
+                time.sleep(1)
+
+                # reset state
+                # TODO check if subgrid needs to be reset
+                previous_subgrid = state.user_select_subgrid
+                state = UserTermState()
+
+                # check if subgrid needs to be updated
+                if state.save_subgrid:
+                    state.user_select_subgrid = previous_subgrid
+                    state.subgrid_select = False
+
+                # flip the player turn
+                if player_active == 1:
+                    player_active = 2
+                elif player_active == 2:
+                    player_active = 1
+
+                term_info = [""] * 3
+                term_info[0] = f"Player {player_active} Active"
+                term_info[1] = "Ready? press y"
+
+                print(term.move_up(7))
+                user_section = update_user_section(term_info)
+                print("".join(user_section))
+
+    print(term.clear)
```

Comment:
This is unnecessary, `term.fullscreen()` clears the screen when you exit fullscreen mode.